cfgov/core/views.py

```diff
@@ -46,3 +52,85 @@ def govdelivery_subscribe(request):
                                                          question_id,
                                                          answer_text)
     return passing_response
+
+
+REQUIRED_PARAMS_REGSGOV = [
+    'comment_on',
+    'general_comment',
+    'first_name',
+    'last_name'
+]
+
+
+@csrf_exempt
+@require_http_methods(['POST'])
+def regsgov_comment(request):
+    """
+    View that checks to see if the request is AJAX, attempts to submit
+    the comment, then either redirects to an error/success page (non-AJAX) or
+    in the case of AJAX, returns some JSON to tell the front-end.
+    """
+    is_ajax = request.is_ajax()
+    if is_ajax:
+        failing_response = JsonResponse({'result': 'fail'})
+    else:
+        failing_response = redirect('reg_comment:server_error')
+
+    for required_param in REQUIRED_PARAMS_REGSGOV:
+        if not request.POST.get(required_param):
+            return failing_response if is_ajax \
+                else redirect('reg_comment:user_error')
+
+    try:
+        submission_response = submit_comment(request.POST)
+        if submission_response.status_code != 201:
+            return failing_response
+    except Exception:
+        return failing_response
+
+    json_data = json.loads(submission_response.text)
+    tracking_number = json_data.get('trackingNumber')
+
+    # For non-ajax, tracking number will appear as a message with tag: success
+    messages.success(request, tracking_number)
+
+    return JsonResponse(
+        {'result': 'pass', 'tracking_number': tracking_number}
+    ) if is_ajax else redirect('reg_comment:success')
+
+
+REGSGOV_BASE_URL = os.environ.get('REGSGOV_BASE_URL')
+REGSGOV_API_KEY = os.environ.get('REGSGOV_API_KEY')
```

Comment:
I'll defer to @rosskarchner @richaagarwal @kave and @kurtw, but I think getting these values from the environ should happen in settings, rather than in the view, and the view should just look for `settings.REGSGOV_BASE_URL` and `settings. REGSGOV_API_KEY`.

That way if anything related to the environment ever changes for whatever reason, we just have to touch the settings file, not views too.